### PR TITLE
MGMT-16691: Document OSImageAdditionalParamsRef functionality for infrastructure operator

### PR DESCRIPTION
**Pull request by paul-maidment**
This is documentation to describe the feature that allows a user to define headers and query params to be used by the image service every time a request to pull an OS image is made.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: docs/operator.md [hunk 1]
```diff
@@ -242,6 +242,75 @@ Add the annotation to the AgentServiceConfig:
 oc annotate --overwrite AgentServiceConfig agent unsupported.agent-install.openshift.io/assisted-image-service-skip-verify-tls=true
 ```
 
+### Custom headers and query parameters for Assisted Image Service OS Image Download
+
+It is possible to specify custom headers and query parameters to be used when downloading OS images.
+
+A config map to contain the headers and query parameters to be used should be created.
+The easiest appreoach is to create up to two files as required, one file to hold any headers, the other to hold query parameters.
```
Comment (CrystalChun) [inline docs/operator.md:250]:
```suggestion
The easiest approach is to create up to two files as required, one file to hold any headers, the other to hold query parameters.
```

**Review by CrystalChun — COMMENTED** [review]
Looks good, just a question (sorry didn't see the code changes): 
Can the user create only a header file or only a query file? 

If so, how do we determine a user provides only a header file? Like do we have some way of detecting or distinguishing it?

If not, can we have something here that says both files are required?

**Comment by paul-maidment** [comment]
> Looks good, just a question (sorry didn't see the code changes): Can the user create only a header file or only a query file?
> 
> If so, how do we determine a user provides only a header file? Like do we have some way of detecting or distinguishing it?
> 
> If not, can we have something here that says both files are required?

The user may provide either/or a header or a query file, we place no restriction on this.
There are some scenarios where I imagine a user may want to use both, technically speaking the user doesn't even have to be trying to achieve authorization, they may simply want to add a query param or a specific header for some reason.

File: docs/operator.md [hunk 2]
```diff
@@ -242,6 +242,75 @@ Add the annotation to the AgentServiceConfig:
 oc annotate --overwrite AgentServiceConfig agent unsupported.agent-install.openshift.io/assisted-image-service-skip-verify-tls=true
 ```
 
+### Custom headers and query parameters for Assisted Image Service OS Image Download
+
+It is possible to specify custom headers and query parameters to be used when downloading OS images.
+
+A config map to contain the headers and query parameters to be used should be created.
+The easiest approach is to create up to two files as required, one file to hold any headers, the other to hold query parameters.
+Use these files to create a secret. The files may be discarded after creation of the secret.
```
Comment (CrystalChun) [inline docs/operator.md:251]: Do they create a secret or configmap?

File: docs/operator.md [hunk 3]
```diff
@@ -242,6 +242,75 @@ Add the annotation to the AgentServiceConfig:
...
+         	"param1": "value1",
+         	"param2": "value2",
+         }
+       EOF
+       
+       oc create secret generic -n multicluster-engine os-images-http-auth --from-file=./query_params --from-file=./headers
+```
+
+Then in the `AgentServiceConfig`, this Secret should be referenced, in `OSImageAdditionalParamsRef` 
+The headers and query params contained within the ConfigMap referenced by `OSImageAdditionalParamsRef` will then be used when pulling osImages.
```
Comment (CrystalChun) [inline docs/operator.md:271]: Same here, should it be a ConfigMap or a Secret?

Comment (paul-maidment) [reply docs/operator.md]: agreed. It's a Secret and not a ConfigMap, I have updated both locations to reflect

**Review by CrystalChun — APPROVED** [review]
/lgtm

**Pull request merged**